Upload only support

mliudev opened the pull request:
Add a flag to limit download capability.

mliudev committed:
Update npm packages.

`npm audit` showed several security issues that required updating
packages. This commit updates a large number of dependencies however
there are still 5 unresolved vulnerabilities (3 low, 2 moderate) which
require updating dependencies of the following dependencies:

* browser-sync
* browser-sync-spa
* karma-angular-filesort

I've elected to ignore these sub-dependencies since the above
dependencies are already updated to their latest versions.

mliudev committed:
Create an allowDownload binding for the bucket controller.

This flag indicates whether to allow clicking on files for downloading.
'allowDownload' provides support for ingest only workflows.

mliudev committed:
Generate download links based on allowDownload bucket flag.

If allowDownload is false then we do not generate a download link and
subsequently do not generate an anchor link for the file.

mliudev committed:
Remove minimize option from ExtractTextPluginLoader.

This option has been removed in the latest version citing semantic
reasons such as ExtractTextPluginLoader should not handle css
uglification.

mliudev committed:
Fix initialization of allowDownload.

In certain situations the constructor overwrites the binding instead of
the other way around. Initializing bindings in $onInit() is a more
reliable way of doing this.

mliudev committed:
Create a build:dev gulp target for builds with sourcemaps.

`gulp build:dev` creates a build with sourcemaps into the dev/
directory. This is useful for debugging in the browser.

mliudev committed:
Add `allow-download` attribute to the bucket directive in s3session.html

This doesn't need to be explicitly set but it indicates to the developer
that this is an available option.

mliudev commented:
![](https://github.trello.services/images/mini-trello-icon.png) [Modify S3Commander to use a flag to toggle download access.](https://trello.com/c/a3EljO0C/6-modify-s3commander-to-use-a-flag-to-toggle-download-access)

CtrlC-Root reviewed (changes requested):
Please modify the HTML template to expose `allow-download` as a checkbox. This makes testing easier and demos the feature to someone looking at the sample. Otherwise code looks good.

mliudev committed:
Fix webpack configuration for serving the build.

This broke when creating the `build:dev` gulp target since both dev and
serve both reference the same `webpack.conf.js` configuration file. This
commit reverts the change so that files are compiled into the .tmp
folder where the `serve` target expects them to be.

mliudev committed:
Create a new webpackDevConf configuration.

This conf file configures webpack to use the correct dev paths for
generating the `build:dev` gulp target. It fixes the issue of compiling
into the `.tmp` folder which should only be used for serving with
browsersync.

mliudev committed:
Add a checkbox for toggling downloadable files to s3 session.

The checkbox should help with testing and demos of the feature to toggle
download support for files.

mliudev commented:
@CtrlC-Root I've added the requested checkbox toggle. Ready for re-review.
